lib/spack/llnl/util/filesystem.py

```diff
@@ -300,6 +306,32 @@ def change_sed_delimiter(old_delim, new_delim, *filenames):
         filter_file(double_quoted, '"%s"' % repl, f)
 
 
+@system_path_filter
+def get_owner_uid(path, err_msg=None):
```

Comment:
It's not clear to me why this was moved out of lib/spack/spack/util/path.py to here?

Reply:
This method requires the import of `llnl.util.filesystem` into `lib/spack/spack/util/path.py` so I can utilize functions from the path module here. Further, between the path, paths, filesystem, executable, and a few other modules it was very easy to get into a circular import, so I refactored the path module to have as minimal dependencies as possible. Moving this allows us to import the path module into the executable module, so we can take advantage of the path normalizing.

Reply:
Makes sense. Thanks for the explanation.